### PR TITLE
moving fairshare method from sched to fairshare

### PR DESCRIPTION
**Pull request by birbalsain**
#### Describe Bug or Feature
Moving fairshare related methods to Fairshare class from Scheduler  class

#### Describe Your Change
Moved  fairshare related methods to Fairshare class from Scheduler  class

#### Attach Test and Valgrind Logs/Output

Description: Rerun All Tests From #7823
Platforms: UBUNTU1804,SUSE15,UBUNTU1604,CENTOS8,RHEL8,CENTOS7,RHEL7,SLES12,SLES15,UBUNTU2004,WIN2K16
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|7831|3374|2|0|0|0|3372|

### CODE REVIEW

**Review by shilpakodli — COMMENTED** [review]
I see methods are still present in pbs_sched.py file. Also there is a revert_fairshare in revert_to_defaults in pbs_sched.py which should be changed to self.fairshare.revert_fairshare. Applies to other methods too.

File: test/fw/ptl/lib/ptl_fairshare.py [hunk 1]
```diff
@@ -267,3 +269,180 @@ def __str__(self):
...
+        self.user = user
+
+    def revert_fairshare(self):
+        """
+        Helper method to revert scheduler's fairshare tree.
+        """
+        cmd = [os.path.join(self.pbs_conf['PBS_EXEC'], 'sbin', 'pbsfs'), '-e']
+        if self.sc_name is not 'default':
+            cmd += ['-I', self.sc_name]
+        self.du.run_cmd(cmd=cmd, runas=self.user)
```
Comment (shilpakodli) [inline test/fw/ptl/lib/ptl_fairshare.py:298]: Include self.hostname in run_cmd.

**Comment by birbalsain** [comment]
closing to restart one failed checks

**Review by shilpakodli — APPROVED** [review]
LGTM

**Review by shilpakodli — CHANGES_REQUESTED** [review]
Hi @birbalsain I see there are few tests which use the methods which moved to fairshare:
test/tests/functional/pbs_backfill_around_strictordering_ext.py
test/tests/functional/pbs_fairshare_jobsortkeys_ext.py
test/tests/functional/pbs_eligible_time_ext.py

**Comment by birbalsain** [comment]
> Hi @birbalsain I see there are few tests which use the methods which moved to fairshare:
> test/tests/functional/pbs_backfill_around_strictordering_ext.py
> test/tests/functional/pbs_fairshare_jobsortkeys_ext.py
> test/tests/functional/pbs_eligible_time_ext.py

yes, Extension PR is ready , will raise after getting tests results

**Comment by birbalsain** [comment]
Description: Rerun All Tests From #7831
Platforms: UBUNTU1804,SUSE15,UBUNTU1604,CENTOS8,RHEL8,CENTOS7,RHEL7,SLES12,SLES15,UBUNTU2004,WIN2K16
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|7842|3374|4|0|0|0|3370|


Description: Rerun Only Failed Tests From #7842
Platforms: UBUNTU1804,SUSE15,UBUNTU1604,CENTOS8,RHEL8,CENTOS7,RHEL7,SLES12,SLES15,UBUNTU2004
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|7844|4|0|0|0|0|4|

**Review by shilpakodli — APPROVED** [review]

**Review by prakashcv13 — APPROVED** [review]

**Pull request merged**